sos/policies/distros/__init__.py

```diff
@@ -762,6 +878,63 @@ def upload_ftp(self, url=None, directory=None, user=None, password=None):
         except IOError:
             raise Exception("could not open archive file")
 
+    def upload_s3(self, endpoint=None, region=None, bucket=None, prefix=None,
+                  access_key=None, secret_key=None):
+        """Attempts to upload the archive to an S3 bucket.
+
+        :param endpoint: The S3 endpoint to upload to
+        :type endpoint: str
+
+        :param region: The S3 region to upload to
+        :type region: str
+
+        :param bucket: The name of the S3 bucket to upload to
+        :type bucket: str
+
+        :param prefix: The prefix for the S3 object/key
+        :type prefix: str
+
+        :param access_key: The access key for the S3 bucket
+        :type access_key: str
+
+        :param secret_key: The secret key for the S3 bucket
+        :type secret_key: str
+
+        :returns: True if upload is successful
+        :rtype: bool
+
+        :raises: Exception if upload is unsuccessful
+        """
+
+        if not endpoint:
+            endpoint = self.get_upload_s3_endpoint()
+        if not region:
+            region = self.get_upload_s3_region()
+
+        if not bucket:
+            bucket = self.get_upload_s3_bucket()
+
+        if not prefix:
+            prefix = self.get_upload_s3_object_prefix()
+
+        if not access_key:
+            access_key = self.get_upload_s3_access_key()
+
+        if not secret_key:
+            secret_key = self.get_upload_s3_secret_key()
+
+        s3_client = boto3.client('s3', endpoint_url=endpoint,
```

Comment:
You probably want to have a condition here with `BOTO3_LOADED`. As if the import hasn't work above, this is going to throw an error. Especially if we're going to shift the boto3 package in Recommends or something like that.

So, I would throw an exception, and then catch it below, and suggest that for this work we need `python3-boto` installed, and then we can let the admin/user decide